flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -0,0 +1,202 @@
+	//  Source Options
+	// ------------------------------------------------------------------------
+	public static final ConfigOption<String> SQL_EXEC_SOURCE_IDLE_TIMEOUT =
+			key("sql.exec.source.idle.timeout")
+					.defaultValue("-1 ms")
+					.withDescription("When a source do not receive any elements for the timeout time, " +
+							"it will be marked as temporarily idle. This allows downstream " +
+							"tasks to advance their watermarks without the need to wait for " +
+							"watermarks from this source while it is idle.");
+
+	// ------------------------------------------------------------------------
+	//  Sort Options
+	// ------------------------------------------------------------------------
+	public static final ConfigOption<Integer> SQL_EXEC_SORT_DEFAULT_LIMIT =
+			key("sql.exec.sort.default.limit")
+					.defaultValue(200)
+					.withDescription("Default limit when user don't set a limit after order by. ");
+
+	public static final ConfigOption<Integer> SQL_EXEC_SORT_MAX_NUM_FILE_HANDLES =
+			key("sql.exec.sort.max-num-file-handles")
+					.defaultValue(128)
+					.withDescription("The maximal fan-in for external merge sort. It limits the number of file handles per operator. " +
+							"If it is too small, may cause intermediate merging. But if it is too large, " +
+							"it will cause too many files opened at the same time, consume memory and lead to random reading.");
+
+	public static final ConfigOption<Boolean> SQL_EXEC_SORT_ASYNC_MERGE_ENABLED =
+			key("sql.exec.sort.async-merge.enabled")
+					.defaultValue(true)
+					.withDescription("Whether to asynchronously merge sorted spill files.");
+
+	// ------------------------------------------------------------------------
+	//  Spill Options
+	// ------------------------------------------------------------------------
+	public static final ConfigOption<Boolean> SQL_EXEC_SPILL_COMPRESSION_ENABLED =
+			key("sql.exec.spill.compression.enabled")
+					.defaultValue(true)
+					.withDescription("Whether to compress spilled data. " +
+							"(Now include sort and hash agg and hash join)");
+
+	public static final ConfigOption<String> SQL_EXEC_SPILL_COMPRESSION_CODEC =
+			key("sql.exec.spill.compression.codec")
+					.defaultValue("lz4")
+					.withDescription("Use that compression codec to compress spilled file. " +
+							"Now we only support lz4.");
+
+	public static final ConfigOption<Integer> SQL_EXEC_SPILL_COMPRESSION_BLOCK_SIZE =
+			key("sql.exec.spill.compression.block-size")
+					.defaultValue(64 * 1024)
+					.withDescription("The buffer is to compress. The larger the buffer," +
+							" the better the compression ratio, but the more memory consumption.");
+
+	// ------------------------------------------------------------------------
+	//  Resource Options
+	// ------------------------------------------------------------------------
+
+	public static final ConfigOption<Integer> SQL_RESOURCE_DEFAULT_PARALLELISM =
```

Comment:
All options in this class should be prefixed with `table.exec.`. If not, move them into a separate class?

Reply:
+1 to `.exec.resource`